java/test/src/main/java/io/ray/test/ObjectStoreTest.java

```diff
@@ -37,4 +69,26 @@ public void testGetMultipleObjects() {
     List<ObjectRef<Integer>> refs = ints.stream().map(Ray::put).collect(Collectors.toList());
     Assert.assertEquals(ints, Ray.get(refs));
   }
+
+  @Test
+  public void testPutWithAssignedOwner() throws Exception {
+    try {
+      ActorHandle<Creator> creator = Ray.actor(Creator::new).remote();
+      ActorHandle<Owner> owner = Ray.actor(Owner::new).remote();
+      ActorHandle<Borrower> borrower = Ray.actor(Borrower::new).remote();
+      Ray.get(owner.task(Owner::warmup).remote());
+      ObjectRef<ObjectRef<Integer>> ref = creator.task(Creator::put, 1, owner).remote();
+      Ray.get(owner.task(Owner::set, ref).remote());
+      creator.task(Creator::exit).remote();
+      Thread.sleep(10000);
+      int data = Ray.get(borrower.task(Borrower::get, ref).remote());
+      Assert.assertEquals(data, 1);
+    } catch (RayTaskException e) {
+      // This will be thrown in local mode test
+      // otherwise it should pass
+      if (!(e.getCause() instanceof IllegalArgumentException)) {
```

Comment:
You don't need this kind of hack to pass the local mode test. You just need to add `@Test(groups = {"cluster"})` on the test method/class and it will be skipped in the local mode test. BTW, shouldn't it be `UnsupportedOperationException` rather than `IllegalArgumentException`?